src/editor/commentwidget.cpp

```diff
@@ -0,0 +1,219 @@
+#include "commentwidget.h"
+#include "ui_commentwidget.h"
+#include<QDebug>
+
+CommentWidget::CommentWidget(QWidget *parent) :
+    QWidget(parent),
+    ui(new Ui::CommentWidget)
+{
+    ui->setupUi(this);
+    commentTabWIdget = new QTabWidget;//create the new tabwidget where
+
+    //first tab will be for editing text, setting bold/italic text, writing, etc.
+    editTab = new AddCommentTextEdit;
+    QVBoxLayout *lay1 = new QVBoxLayout;
+    lay1->addWidget(editTab);
+    QWidget *tab1 = new QWidget;
+    tab1->setLayout(lay1);
+
```

Comment:
Here you wrap AddCommentTextEdit in another widget (tab1) 
What is the purpose?
I believe it is unnecessary.
You might just write: 
commentTabWIdget->addTab(editTab, "Edit");

The same remark concerning tab2

Reply:
done